CONTRIBUTING.rst

```diff
@@ -162,3 +162,150 @@ their individual contributions.
 * `Will Hall <https://www.github.com/wrhall>`_ (`wrsh07@gmail.com <mailto:wrsh07@gmail.com>`_)
 * `Will Thompson <https://www.github.com/wjt>`_ (`will@willthompson.co.uk <mailto:will@willthompson.co.uk>`_)
 * `Zac Hatfield-Dodds <https://www.github.com/Zac-HD>`_ (`zac.hatfield.dodds@gmail.com <mailto:zac.hatfield.dodds@gmail.com>`_)
+
+
+-----------------
+Review Guidelines
+-----------------
+
+All changes to Hypothesis go through a code review process, and must be
+signed off by at least one person with write access to the repo other
+than the author of the change. This document
+outlines some guidelines on what to look for and what we're interested
+in. It applies uniformly to maintainers and external contributors.
+
+At a high level, the two things we're looking for in review are answers
+to the following questions:
+
+1. Is this change going to make users' lives worse?
+2. Is this change going to make the maintainers' lives worse?
+
+Code review is a collaborative process between the author and the
+reviewer to try to ensure that the answer to both of those questions
+is no.
+
+Ideally of course the change should also make one or both of the users'
+and our lives *better*, but it's OK for changes to be mostly neutral.
+The author should be presumed to have a good reason for submitting the
+change in the first place, so neutral is good enough!
+
+The rest of this section outlines specific things reviewers should
+focus on in aid of this, broken up by sections according to their area
+of applicability.
+
+All of these conditions must be satisfied for merge. Where the reviewer
+thinks this conflicts with the above higher level goals, they may make
+an exception if both the author and another maintainer agree.
+
+~~~~~~~~~~~~~~~~~
+General Questions
+~~~~~~~~~~~~~~~~~
+
+The following are required for almost every change:
+
+1. Changes must be of reasonable size. If a change could logically
+   be broken up into several smaller changes that could be reviewed
+   separately on their own merits, it should be.
+2. The motivation for each change should be clearly explained (this
+   doesn't have to be an essay, especially for small changes, but
+   at least a sentence of explanation is usually required).
+3. The likely consequences of a change should be outlined (again,
+   this doesn't have an essay, and it may be sufficiently
+   self-explanatory that the motivation section is sufficient).
+
+~~~~~~~~~~~~~~~~~~~~~
+Functionality Changes
+~~~~~~~~~~~~~~~~~~~~~
+
+This section applies to any changes in Hypothesis's behaviour, regardless
+of their nature.
+
+1. The code should be clear in its intent and behaviour.
+2. Behaviour changes should come with appropriate tests to demonstrate
+   the new behaviour.
+3. Hypothesis must never be *flaky*. Flakiness here is
+   defined as anything where a test fails and this does not indicate
+   a bug in Hypothesis or in the way the user wrote the code or the test.
```

Comment:
How would this apply to a (hypothetical) pull request adding the assume() function?
There's some probability that assume() fails to find enough examples via random chance.